### PR TITLE
20231017 volume

### PR DESCRIPTION
**Pull request by mau-fyi**
Adds two api endpoints for getting and setting volume

### CODE REVIEW

File: AudioStream.js [hunk 1]
```diff
@@ -16,6 +19,8 @@ module.exports = class AudioStream {
     this.paused = false;
     this.ytdl = null;
     this.mpv = null;
+    this.volume = INITIAL_VOLUME;
+    this.setVolume(this.volume);
```
Comment (evanugarte) [inline AudioStream.js]: `setVolume` modifies `this.volume`
```suggestion
    this.setVolume(this.volume);
```

File: api.js [hunk 2]
```diff
@@ -19,6 +19,24 @@ app.post('/stream', bodyParser.json(), async (req, res) => {
...
 });
 
+/**
+ * /setVolume post request expects the following JSON body:
+ * @param {number} volume - The volume to set the stream to. Must be between 0 and 100.
+ */
+app.post('/setVolume', bodyParser.json(), async (req, res) => {
+  try {
+    const volume = req.body.volume;
+    audioStream.setVolume(volume);
```
Comment (evanugarte) [inline api.js:35]: are you sure `volume` is a number here and not a string? might want to cast it to a number before calling `setVolume`

also might want to check that the value is a number and not `"asdf"`, you can do this with `Number.isNaN`
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/isNaN

File: api.js [hunk 3]
```diff
@@ -19,6 +19,24 @@ app.post('/stream', bodyParser.json(), async (req, res) => {
...
+    res.status(200).json({ volume: volume });
+  } catch (error) {
+    res.status(500).json({ error: 'An error occurred while setting volume. Please make sure that volume is between 0 and 100.' });
+  }
+});
+
+app.get('/volume', async (req, res) =>
+  res.status(200).json({ volume: audioStream.getVolume() })
```
Comment (evanugarte) [inline api.js]: we can just do res.json, it implies the response is 200

https://stackoverflow.com/a/58401939

File: AudioStream.js (hunk 1 above)
Comment (mau-fyi) [reply AudioStream.js]: done

File: api.js (hunk 2 above)
Comment (mau-fyi) [reply api.js:35]: Done and moved valid number check to api.js plus added more types of error responses

File: api.js (hunk 3 above)
Comment (mau-fyi) [reply api.js]: added change to entire file

**Pull request closed without merging**